### ISSUE TITLE
No compile-time errors when using templates

### ISSUE THREAD

**Issue by rabacadabra** [casid/jte#194]
I have an `openForm.jte` file which opens a `<form>` tag and adds some hidden fields:
```
<%--Opens a form tag--%>
<form action="${ShortbookPage.Companion.getEndpoint()}" method="get">

<%--Adds some hidden fields--%>
@template.common.params(paramsMap = page.getData().getParams.getParamMap(List.of("search")))
```

I have another `endForm.jte` template which only closes the `<form>`:
```
</form>
```

Both templates are used in the page's jte file this way:
```
@template.common.openForm(page = page)

<input type="text" name="search" placeholder="Enter search keyword"
       value="${page.getData().getParams.search}">

@template.common.endForm()
```

This is done this way so that if at all there are any form params which are added later on, they get added automatically without having to add an additional line in the JTE file.

Issue:
If I miss adding the call to `endForm()`, the project compiles successfully. If i'm using the <form> and </form> directly in the page, and the closing tag is missed, it does throw a compile-time error.

Is this an issue or is it expected when using templates?

**Comment by casid** [comment]
This is a feature of jte in `Html` mode, to check that HTML tags inside a template are properly closed.

I would recommend to use a `gg.jte.Content` block, this way only one template is needed and as a benefit closing the form cannot be forgotten by the user. For example:

`form.jte`
```html
@import gg.jte.Content

@param Page page
@param Content content

<form action="${...}">
   <%--Adds some hidden fields--%>
   @template.common.params(paramsMap = page.getData().getParams.getParamMap(List.of("search")))

   ${content}
</form>
```

`page.jte`:
```html
@template.common.form(page = page, content = @`
   <input type="text" name="search" placeholder="Enter search keyword"
       value="${page.getData().getParams.search}">
`)
```

**Comment by rabacadabra** [comment]
I think my issue wasn't clear enough. when the <form> and </form> are inserted into the page from templates, how come there is no compile-time error? aren't the templates processed during compilation?

`Content` seems like a good option, thanks for the suggestion!

**Comment by rabacadabra** [comment]
what i want to know why this gets compiled:
```
@template.common.openForm(page = page)
<input type="text" name="search" placeholder="Enter search keyword" value="${page.getData().getParams.search}">
<%--@template.common.endForm()--%>
```

when this doesn't:
```
<form action="/shortbook" method="get">
    <input type="text" name="search" placeholder="Enter search keyword" value="${page.getData().getParams.search}">
<%--</form>--%>
```

Note that the form's closing tag is missing in both cases.

**Comment by casid** [comment]
Ah, that's because the check is currently limited per template, as each template can be compiled separately.

**Comment by kelunik** [comment]
But openForm should fail to compile, no?

**Comment by rabacadabra** [comment]
openForm and endForm seems to be compiling successfully

**Comment by casid** [comment]
@kelunik you are right, those should not compile.

**Comment by casid** [comment]
This is fixed an will be part of jte 3.0
